File: UaicLibraryWebSpa/src/app/models/book.ts
import { Author } from './author';
import { BookComment } from './book-comment.model';
import { BookCategory } from './book-category';
import { Faculty } from './faculty';

export class Book {
    constructor(
        public id: number,
        public isbn: string,
        public bookAuthors: Array<Author>,
        public description: string,
        public name: string,
        public numberOfPages: number,
        public comments: Array<BookComment>,
        public bookCategory: BookCategory,
        public publishDate: Date,
        public likes: number,
        public dislikes: number,
        public faculty: Faculty,
        public bookUrl: string,
        public views: number,
        public reads: number,
        public imageUrl: string,
        public pageAnnotations: number
    ) { }
}

File: UaicLibrary.Domain/BookManagement/Book.cs
using System;
using System.Collections.Generic;
using UaicLibrary.Domain.AuthorManagement;
using UaicLibrary.Domain.BookCategoryManagement;
using UaicLibrary.Domain.BookCommentManagement;
using UaicLibrary.Domain.FacultyManagement;

namespace UaicLibrary.Domain.BookManagement
{
    public class Book
    {
        public int Id { get; set; }
        public string Isbn { get; set; }
        public IList<Author> BookAuthors { get; set; }
        public string Description { get; set; }
        public string Name { get; set; }
        public int NumberOfPages { get; set; }
        public IList<BookComment> Comments { get; set; }
        public BookCategory BookCategory { get; set; }
        public DateTime PublishDate { get; set; }
        public string ImageUrl { get; set; }
        public  int Likes { get; set; }
        public  int Dislikes { get; set; }
        public Faculty Faculty { get; set; }
        public string BookUrl { get; set; }
        public int Views { get; set; }
        public int Reads { get; set; }
        public int PageAnnotations { get; set; }
    }
}
